core/quorum_initiator.go

```diff
@@ -656,29 +666,55 @@ func (c *Core) finishConsensus(id string, qt int, p *ipfsport.Peer, status bool,
 	defer c.qlock.Unlock()
 	cs, ok := c.quorumRequest[id]
 	if !ok {
+		fmt.Println("failed to get quorum consensus")
+		if p != nil {
+			p.Close()
+		}
+		return
+	}
+	pd, ok := c.pd[id] //getting details of quorums who pledged
+	if !ok {
+		fmt.Println("failed to get pledged token details")
 		if p != nil {
 			p.Close()
 		}
 		return
 	}
+	pledgingQuorumDID := make([]string, 0, len(pd.PledgedTokens))
+	for k := range pd.PledgedTokens {
+		pledgingQuorumDID = append(pledgingQuorumDID, k)
+	}
+	var pledgingDID string
+	if len(pledgingQuorumDID) > 0 {
+		pledgingDID = pledgingQuorumDID[0]
+	}
+
 	switch qt {
 	case 0:
 		cs.Result.RunningCount--
 		if status {
 			did := p.GetPeerDID()
-			if cs.Result.SuccessCount < MinConsensusRequired {
-				csig := CreditSignature{
-					Signature:     util.HexToStr(ss),
-					PrivSignature: util.HexToStr(ps),
-					DID:           did,
-					Hash:          hash,
-				}
+			csig := CreditSignature{
+				Signature:     util.HexToStr(ss),
+				PrivSignature: util.HexToStr(ps),
+				DID:           did,
+				Hash:          hash,
+			}
+			if cs.Result.SuccessCount < MinConsensusRequired-1 {
+				cs.P[did] = p
+				cs.Credit.Credit = append(cs.Credit.Credit, csig)
+				cs.Result.SuccessCount++
+			} else if did == pledgingDID && cs.Result.SuccessCount == MinConsensusRequired-1 {
+				cs.P[did] = p
+				cs.Credit.Credit = append(cs.Credit.Credit, csig)
+				cs.Result.SuccessCount++
+			} else if cs.Result.RunningCount == 0 && cs.Result.SuccessCount == MinConsensusRequired-1 {
```

Comment:
In this case , with 7 nodes setup and running , aren't you counting vote of last (7th) node to respond instead of the 5th node?  5th and 6th nodes response will be discarded since RunningCount will not be 0.  Hence waiting time is more and early responders don't get credit.

Reply:
This can be resolved. For the above scenario, we would need to perform a few more checks and probably need one more variable. So, that is a tradeoff here. Also, since the process would get more complicated when equal pledging is implemented with type1 quorums and the above algorithm will change for type1. And in type2, we are using our own quorums and order of their signature will not be fixed, so I was thinking if we can bear this tradeoff. Please let me know, if the changes need to be made, I will do it right away.

Reply:
We can maybe create a list of dids that pledged and check if the signing did is on that list or not. Or maybe a map that will hold the pledging quorum did, with did as the key and value will be a boolean value. Whenever a pledging did signs, we can update the boolean value to True. Using a list would increase the number of iterations. I think a map would be better in this case.